### ISSUE TITLE
File '/tmp/sqlflow227199809/input.sql' cannot be read

### ISSUE THREAD

**Issue by BlackPoint-CX** [sql-machine-learning/sqlflow#1227]
**Description**
It seems that SQLFlow cannot read temporary file saved on server.

**Reproduction Steps**
After restart the kernel of notebook, the following exception will be thrown out.

```
_Rendezvous: <_Rendezvous of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "thirdPartyParse failed: java.io.IOException: File '/tmp/sqlflow227199809/input.sql' cannot be read
	at org.apache.commons.io.FileUtils.openInputStream(FileUtils.java:294)
	at org.apache.commons.io.FileUtils.readFileToByteArray(FileUtils.java:1851)
	at org.sqlflow.parser.ParserAdaptorCmd.main(ParserAdaptorCmd.java:42)
 exit status 255"
	debug_error_string = "{"created":"@1574340457.693353044","description":"Error received from peer ipv4:10.82.128.7:8005","file":"src/core/lib/surface/call.cc","file_line":1055,"grpc_message":"thirdPartyParse failed: java.io.IOException: File '/tmp/sqlflow227199809/input.sql' cannot be read\n\tat org.apache.commons.io.FileUtils.openInputStream(FileUtils.java:294)\n\tat org.apache.commons.io.FileUtils.readFileToByteArray(FileUtils.java:1851)\n\tat org.sqlflow.parser.ParserAdaptorCmd.main(ParserAdaptorCmd.java:42)\n exit status 255","grpc_status":2}"
>
```

**Expected Behavior**
What you expected to happen.

**Screenshots**

**Environment (Please complete the following information):**
 - OS:
 - Browser:
 - Version:

**Additional Notes**

**Comment by typhoonzero** [comment]
In my case, I've encountered an error:

```sql
%%sqlflow
SELECT *
FROM iris.train
TO TRAIN DNNClassifier
WITH model.n_classes = 3,
  model.hidden_units = [10, 10],
  train.epoch = 100
COLUMN sepal_length, sepal_width, petal_length, petal_width
LABEL class
INTO sqlflow_models.my_dnn_model;
```


```
_Rendezvous: <_Rendezvous of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "thirdPartyParse failed: ERROR StatusLogger Unrecognized format specifier [d]
ERROR StatusLogger Unrecognized conversion specifier [d] starting at position 16 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [thread]
ERROR StatusLogger Unrecognized conversion specifier [thread] starting at position 25 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [level]
ERROR StatusLogger Unrecognized conversion specifier [level] starting at position 35 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [logger]
ERROR StatusLogger Unrecognized conversion specifier [logger] starting at position 47 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [msg]
ERROR StatusLogger Unrecognized conversion specifier [msg] starting at position 54 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [n]
ERROR StatusLogger Unrecognized conversion specifier [n] starting at position 56 in conversion pattern.
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console. Set system property 'log4j2.debug' to show Log4j2 internal initialization logging.
ERROR StatusLogger Unrecognized format specifier [d]
ERROR StatusLogger Unrecognized conversion specifier [d] starting at position 16 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [thread]
ERROR StatusLogger Unrecognized conversion specifier [thread] starting at position 25 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [level]
ERROR StatusLogger Unrecognized conversion specifier [level] starting at position 35 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [logger]
ERROR StatusLogger Unrecognized conversion specifier [logger] starting at position 47 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [msg]
ERROR StatusLogger Unrecognized conversion specifier [msg] starting at position 54 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [n]
ERROR StatusLogger Unrecognized conversion specifier [n] starting at position 56 in conversion pattern.
NoViableAltException(312@[205:64: ( ( KW_AS )? alias= identifier )?])
	at org.antlr.runtime.DFA.noViableAlt(DFA.java:158)
	at org.antlr.runtime.DFA.predict(DFA.java:116)
	at org.apache.hadoop.hive.ql.parse.HiveParser_FromClauseParser.tableSource(HiveParser_FromClauseParser.java:4171)
	at org.apache.hadoop.hive.ql.parse.HiveParser_FromClauseParser.atomjoinSource(HiveParser_FromClauseParser.java:1600)
	at org.apache.hadoop.hive.ql.parse.HiveParser_FromClauseParser.joinSource(HiveParser_FromClauseParser.java:1854)
	at org.apache.hadoop.hive.ql.parse.HiveParser_FromClauseParser.fromSource(HiveParser_FromClauseParser.java:1527)
	at org.apache.hadoop.hive.ql.parse.HiveParser_FromClauseParser.fromClause(HiveParser_FromClauseParser.java:1370)
	at org.apache.hadoop.hive.ql.parse.HiveParser.fromClause(HiveParser.java:45198)
	at org.apache.hadoop.hive.ql.parse.HiveParser.atomSelectStatement(HiveParser.java:39792)
	at org.apache.hadoop.hive.ql.parse.HiveParser.selectStatement(HiveParser.java:40044)
	at org.apache.hadoop.hive.ql.parse.HiveParser.regularBody(HiveParser.java:39690)
	at org.apache.hadoop.hive.ql.parse.HiveParser.queryStatementExpressionBody(HiveParser.java:38900)
	at org.apache.hadoop.hive.ql.parse.HiveParser.queryStatementExpression(HiveParser.java:38788)
	at org.apache.hadoop.hive.ql.parse.HiveParser.execStatement(HiveParser.java:2396)
	at org.apache.hadoop.hive.ql.parse.HiveParser.statement(HiveParser.java:1420)
	at org.apache.hadoop.hive.ql.parse.ParseDriver.parse(ParseDriver.java:220)
	at org.apache.hadoop.hive.ql.parse.ParseDriver.parse(ParseDriver.java:178)
	at org.apache.hadoop.hive.ql.parse.ParseDriver.parse(ParseDriver.java:173)
	at org.sqlflow.parser.HiveQLParserAdaptor.ParseAndSplit(HiveQLParserAdaptor.java:44)
	at org.sqlflow.parser.ParserAdaptorCmd.main(ParserAdaptorCmd.java:54)
Exception in thread "main" java.lang.StringIndexOutOfBoundsException: String index out of range: 235
	at java.lang.String.substring(String.java:1963)
	at org.sqlflow.parser.HiveQLParserAdaptor.ParseAndSplit(HiveQLParserAdaptor.java:71)
	at org.sqlflow.parser.ParserAdaptorCmd.main(ParserAdaptorCmd.java:54)
```

Seems like a internal parser bug: `Exception in thread "main" java.lang.StringIndexOutOfBoundsException: String index out of range: 235`

cc: @tonyyang-svail 

----

PS: recompile the latest jar and I can run the TRAIN SQL without error.

**Comment by tonyyang-svail** [comment]
@BlackPoint-CX I am not able to reproduce it in `iris-dnn.ipynb`, where I started the Jupyter notebook by `docker run -it -p 8888:8888 sqlflow/sqlflow`.

![Screen Shot 2019-11-21 at 10 38 00 AM](https://user-images.githubusercontent.com/29932814/69366384-20c3ba00-0c4b-11ea-9d9a-7ca7c5a27cab.png)

Could you please check if the file `/tmp/sqlflow227199809/input.sql` exists? If it exists, what is the permission of the file?

**Comment by tonyyang-svail** [comment]
> Seems like a internal parser bug: Exception in thread "main" java.lang.StringIndexOutOfBoundsException: String index out of range: 235

@typhoonzero Are you using the latest Docker image? I believe this error is fixed by #1202.

**Comment by Yancey0623** [comment]
> 	if err := ioutil.WriteFile(inputFile, []byte(sql), 755); err != nil {

change 755 to 0755 can fix this bug.

**Comment by BlackPoint-CX** [comment]
> @BlackPoint-CX I am not able to reproduce it in `iris-dnn.ipynb`, where I started the Jupyter notebook by `docker run -it -p 8888:8888 sqlflow/sqlflow`.
> 
> ![Screen Shot 2019-11-21 at 10 38 00 AM](https://user-images.githubusercontent.com/29932814/69366384-20c3ba00-0c4b-11ea-9d9a-7ca7c5a27cab.png)
> 
> Could you please check if the file `/tmp/sqlflow227199809/input.sql` exists? If it exists, what is the permission of the file?

I had checked the existence of the temp file and `input.sql` but the permission of those were not 755 specified in code. @Yancey1989 and me has found this problem was caused by the format of permission. When change 755 into 0755 can solve this problem.

**Comment by BlackPoint-CX** [comment]
> > Seems like a internal parser bug: Exception in thread "main" java.lang.StringIndexOutOfBoundsException: String index out of range: 235
> 
> @typhoonzero Are you using the latest Docker image? I believe this error is fixed by #1202.

We can not use public docker image in internal environment. Sad.

**Comment by tonyyang-svail** [comment]
> change 755 to 0755 can fix this bug.

@Yancey1989 @BlackPoint-CX Thank you so much for spotting this bug. I realized that we need to write `perm so.FileMode` in octal format, which can be done by prepending a `0` in front of the digit.
